src/tests/ftest/util/yaml_utils.py

```diff
@@ -0,0 +1,469 @@
+                        message = f"Not enough '{key}' placeholder replacements specified"
+                        self.log.error("  - %s; required: %s", message, quantity)
+                        raise YamlException(message) from error
+
+            except ValueError:
+                try:
+                    # Replace clush-style placeholders with nodes from the user provided list using
+                    # a mapping so that values used more than once yield the same replacement
+                    for node in NodeSet(placeholder):
+                        if node not in node_mapping:
+                            try:
+                                node_mapping[node] = replacement.pop(0)
+                            except IndexError as error:
+                                # Not enough nodes provided for the replacement
+                                if not self._override:
+                                    message = f"Not enough '{key}' placeholder replacements remain"
+                                    self.log.error(
+                                        "  - %s; required: %s", message, NodeSet(placeholder))
+                                    raise YamlException(message) from error
+                                break
+                            self.log.debug(
+                                "  - %s replacement node mapping: %s -> %s",
+                                key, node, node_mapping[node])
+                        replacement_nodes.add(node_mapping[node])
+
+                except TypeError as error:
+                    # Unsupported format
+                    message = f"Unsupported placeholder format: {placeholder}"
+                    self.log.error(message)
+                    raise YamlException(message) from error
+
+            hosts_key = r":\s+".join([key, str(placeholder)])
+            hosts_key = hosts_key.replace("[", r"\[")
+            hosts_key = hosts_key.replace("]", r"\]")
+            if replacement_nodes:
+                replacements[hosts_key] = ": ".join([key, str(replacement_nodes)])
+            else:
+                replacements[hosts_key] = None
+
+    @staticmethod
+    def _get_storage_replacement(replacements, placeholder_data, key, replacement):
+        """_summary_.
+
+        Args:
+            replacements (dict): _description_
+            placeholder_data (dict): _description_
+            key (str): _description_
+            replacement (list): _description_
+        """
+        for placeholder in placeholder_data[key]:
+            bdev_key = f"\"{placeholder}\""
+            if bdev_key in replacements:
+                continue
+            try:
+                replacements[bdev_key] = f"\"{replacement.pop(0)}\""
+            except IndexError:
+                replacements[bdev_key] = None
+
+    def _get_timeout_replacement(self, replacements, placeholder_data, key, replacement):
+        """_summary_.
```

Comment:
Do we need summary and descriptions for this method?

Reply:
Yes. Fixed, thanks.